[Maliha Haque] iP

malihahaque opened the pull request:
### Lara

>"For every minute spent organizing, an hour is earned." – Benjamin Franklin

Lara is a smart task manager that helps you stay organized by keeping track of your tasks efficiently.

**Features**
- Text-based interface📝 
-  Easy to use 🚀
-  Fast and efficient⚡

**How to Use**

1. Download the application from [here](https://github.com/malihahaque/ip.git)📥 
2. Double-click to open
3. Add your tasks and manage them easily✍ 
4.  Let Lara handle the rest for you!🎯

**Upcoming Features**

 - [x] Managing tasks

 - [ ] Reminders (coming soon)

**Code Example**
_Here's the main method to launch the application:_

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Lara.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

malihahaque committed:
greetings

malihahaque committed:
greetings

malihahaque committed:
greetings

malihahaque committed:
echo

malihahaque committed:
echo

malihahaque committed:
add,list

malihahaque committed:
add,list

malihahaque committed:
todos, events, deadlines

malihahaque committed:
error handling

malihahaque committed:
error handling

malihahaque committed:
delete

lesterlimjj commented on src/main/java/Lara.java:
```diff
@@ -0,0 +1,122 @@
+
+    private static void printList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i=1; i<=list.size(); i++) {
```
Your operators (e.g. "=", "<=", "-") should be surrounded by space on both sides. This minor issues reappear a few times in other parts of your code(e.g. addTask(String) method).

lesterlimjj commented on src/main/java/Lara.java:
```diff
@@ -0,0 +1,122 @@
+                    unmarkTask(message);
+                } else if (firstWord.equals("delete")) {
+                  deleteTask(message);
+                } else if (firstWord.equals("todo") || firstWord.equals("deadline") || firstWord.equals("event")){
```
Perhaps we can split into more if-else statements and make each task creation its own function?

lesterlimjj commented on src/main/java/LaraException.java:
```diff
@@ -0,0 +1,5 @@
+public class LaraException extends Exception{
```
{ should have a white space preceding it.

lesterlimjj commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+import java.util.ArrayList;
+
+public class Task {
```
Consider putting your object classes into packages (naming for packages need to be all lowercase). It is a week 3 IP requirement but it would be good to start doing it in earlier branches.

lesterlimjj commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected static int taskCounter=0;
```
Perhaps we can remove this if we do not plan to use it?

lesterlimjj commented on src/main/java/Lara.java:
```diff
@@ -0,0 +1,122 @@
+                System.out.println(event);
+            }
+            System.out.println("Now you have " + list.size() + " tasks in the list.");
+        } catch (LaraException e) {
```
Do we need a try-catch for every task-related function call if we already have a try-catch in your if-else statements?

lesterlimjj reviewed (commented):
You are making good progress in catching up. Aside from a few minor coding violations and some long pieces of code, everything else look fine. Before you progress into later weeks of the IP, it would be good to install the checkstyle scripts from week 4 to help you check coding standard violation. Aside from that, configuring your IDE to fit the coding standard is available in week 2 or 3 so perhaps you can skim through and configure it now.

kylieluk88 commented on src/main/java/Lara.java:
```diff
@@ -0,0 +1,122 @@
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println("[ ] " + list.get(taskNumber).getDescription());
+            } catch (LaraException e) {
+            System.out.println(e.getMessage());
```
This line should have another 4-space indentation. some of the spacing for the "}" brackets below also seem a bit inconsistent, can make use of IDEs which automatically adjust the indentation for you so you don't have to manually check :)

kylieluk88 reviewed (commented):
other than some small issues like spacing and indexing, the code is clear and easy to understand! keep it up in the week 3 tasks

malihahaque committed:
testing

malihahaque committed:
testing

malihahaque committed:
Implement Level 8: Add date parsing for Deadlines and Events

malihahaque committed:
Merge Level 8: Dates and Times

malihahaque committed:
date & time

malihahaque committed:
date & time

malihahaque committed:
date & time

malihahaque committed:
A-MoreOOP

malihahaque committed:
A-MoreOOP

malihahaque committed:
A-MoreOOP

malihahaque committed:
A-MoreOOP

malihahaque committed:
A-Packages

malihahaque committed:
A-Gradle

malihahaque committed:
A-JUnit

malihahaque committed:
A-Jar

malihahaque committed:
Added JavaDoc comments to classes

malihahaque committed:
A-JavaDoc

malihahaque committed:
Your commit message

malihahaque committed:
A-CodingStandard

malihahaque committed:
find

malihahaque committed:
Merge branch 'branch-Level-9'

malihahaque committed:
find

malihahaque committed:
find

malihahaque committed:
find

malihahaque committed:
Resolved merge conflicts

malihahaque committed:
Add GUI to Lara chatbot

malihahaque committed:
Add Gui to Lara Chatbot

malihahaque committed:
Add Gui to Lara Chatbot

malihahaque committed:
Add Gui to Lara Chatbot

malihahaque committed:
Add Gui to Lara Chatbot

malihahaque committed:
Use assertions to enforce assumptions

Assertions are added to document important assumptions and ensure program correctness.
These assertions help catch potential issues early in development and debugging.

Changes made:
- Added assertions to validate input parameters in key methods
- Ensured program invariants hold after critical operations
- Improved code reliability by enforcing expected states

Assertions are enabled during development and testing, providing a safeguard against unintended behaviors.

malihahaque committed:
Improve Code Quality: Reduce nesting in start() method

Refactored UI setup into a separate method to improve readability

malihahaque committed:
Merge pull request #2 from malihahaque/branch-A-CodeQuality

Improve code quality and readability

malihahaque committed:
Enhance Gui in Lara Chatbot

malihahaque committed:
Enhance Gui in Lara Chatbot

malihahaque committed:
Merge remote-tracking branch 'origin/master'

malihahaque committed:
Feature add: Prevent duplicate tasks from being added

Implemente duplicate detection in TaskList to prevent redundant entries

malihahaque committed:
Fix bugs in date and time formatting

malihahaque committed:
Add 4 extensions from category C

malihahaque committed:
Add app screenshot

malihahaque committed:
Add app screenshot

malihahaque committed:
Add User Guide

malihahaque committed:
Add User Guide

malihahaque committed:
Add User Guide

malihahaque committed:
Add User Guide

malihahaque committed:
Add User Guide

malihahaque committed:
Fix bugs on saving history

malihahaque committed:
Add User Guide

malihahaque committed:
Add User Guide

malihahaque committed:
Add User Guide

malihahaque committed:
Fix bugs to prepare for release

malihahaque committed:
Fix bugs to prepare for release

malihahaque committed:
Fix bugs to prepare for release

malihahaque committed:
Fix bugs to prepare for release

malihahaque committed:
Fix bugs to prepare for release

malihahaque committed:
Fix bugs to prepare for release

malihahaque committed:
Fix bugs pointed out in the forum during smoke tests

malihahaque committed:
Fix bugs in error messages

malihahaque committed:
Fix bugs in error messages
